examples/air/air.py

```diff
@@ -163,7 +163,8 @@ def prior_step(self, t, n, prev, z_pres_prior_p=default_z_pres_prior_p):
 
     def model(self, data, batch_size, **kwargs):
         pyro.module("decode", self.decode)
-        with pyro.iarange('data', data.size(0), use_cuda=self.use_cuda) as ix:
+        with pyro.iarange('data', data.size(0), subsample_size=batch_size,
```

Comment:
We shouldn't need `subsample_size` annotations in the model; those should be read from the guide via `poutine.replay`.

Reply:
Hmm..let me see if there is a way to get around this.

Reply:
Actually, I'm not sure what's the best way to do this. The issue is that we need the batch shape (`data.shape[0]`) in the model. The only argument provided to `_subsample` in the model is the `size` arg (both `subsample` and `subsample_size` are None). `_Subsample` returns a batch of indices using replay poutine magic which doesn't play well with shape inference in the JIT. Note that this batch size is different from the `size` arg. The only way I see this working is to have `_Subsample.sample` return a tuple of `(indices, batch_size)`. Let me know if you have any other ideas.